workflows/plots.py

```diff
@@ -107,177 +153,84 @@ def plot_all_ne_estimates(sp_infiles, msmc_infiles, gone_infiles, outfile,
-    num_samples = [0 for _ in range(ddb.num_populations)]
-    num_samples[pop_id] = n_samp
-    coal_rate, P = ddb.coalescence_rate_trajectory(steps=steps,
-                                                   lineages=num_samples,  # changed to lineages
-                                                   double_step_validation=False)
-    steps = steps * generation_time
-    return coal_rate, P, steps
 
+def gather_coal_rate(outfile, model, pops_dict, generation_time, steps):
+    header = "method,population,DFE,annotations,year,Ne"
+    with open(outfile[0], 'w') as coal:
+        coal.write(f"{header}\n")
+        for pop_id, pop in enumerate(pops_dict.keys()):
+            coal_rate, P, step_rate, census_size = popn_coal_rate(model, pop, pops_dict, generation_time, steps)
+            for i in range(len(step_rate)):
+                coal.write(f"coal,{pop},none,none,{step_rate[i]},{1/(2*coal_rate[i])}\n")
+                coal.write(f"census,{pop},none,none,{step_rate[i]},{census_size[i][pop_id]}\n")
 
-def plot_sfs(s, outfile):
+
+def plot_compound_Ne_t(infile, outfile, ref_line="coal", colorby="annotations", log=True):
     """
-    Plot the SFS for this simulation
+    figure of N(t) for multiple methods
+    set msmc w/ style="n_samp"
     """
-    bins = [n + 1 for n in range(len(s[0]))]
-    vals = []
-    for i in range(len(s)):
-        vals.append([int(x) for x in s[i]])
-    if len(s) == 2:
-        f, ax = plt.subplots(
-            1, 2, sharey=True, tight_layout=True, figsize=(8, 3))
-        ax[0].bar(bins, vals[0])
-        ax[1].bar(bins, vals[1])
-        ax[0].set_title("all sites")
-        ax[1].set_title("masked sites")
-        ax[0].set_ylabel("counts")
-        ax[0].set_xlabel("derived allele frequency")
-        ax[1].set_xlabel("derived allele frequency")
-        f.savefig(outfile, bbox_inches='tight')
-        plt.close()
-    else:
-        f, ax = plt.subplots(figsize=(3, 3))
-        ax.bar(bins, vals[0])
-        ax.set_title("all sites")
-        ax.set_ylabel("counts")
-        ax.set_xlabel("derived allele frequency")
-        f.savefig(outfile, bbox_inches='tight')
-        plt.close()
-
-
+    # load df
+    df = pd.read_csv(infile, sep=",")
+    df_ddb = pd.read_csv(Path(infile).parent.parent / "coal_estimated_Ne_t.csv", sep=",")
+    df_ddb = df_ddb[df_ddb["method"] == ref_line]
+    # plot params
+    pop_order = np.sort(df['population'].unique())
+    annot_order = np.sort(df[colorby].unique())[::-1]
+    pal_dict = {pop:COLOURS[-(i+1)] for i, pop in enumerate(annot_order)}
+    dfe_order = np.sort(df["DFE"].unique())[::-1]
+    g = sns.relplot(data=df, x="year", y="Ne", col="population", row="DFE",
+                    col_order=list(pop_order), row_order=list(dfe_order),
+                    hue=colorby, hue_order=annot_order,
+                    units="seed", kind="line", palette=pal_dict, 
+                    height=3, aspect=2, estimator=None, errorbar="se", err_style="band",
```

Comment:
I got an error here: `'Line2D' object has no property 'errorbar'`
Commenting out `errorbar="se", err_style="band"` make it run for me and I was able to produce the plot that I think looks like what Scott showed.

Reply:
if you start a new env from scratch it should automatically be at `0.12.2`, although we could pin the version. Did you remove any old environment you had and create a new one?

Reply:
I see. No I didn't create a new one, just updated my existing env using `conda env update --file environment.yml`. I can create a new one now to confirm it would automatically be at `0.12.2`

Reply:
thanks-- let's make sure that works

Reply:
Okay everything works now. The env created fresh did pull the `0.12.2` version for seaborn. And the `errorbar` line of code works for this version.

Reply:
Great!